core/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -140,16 +139,9 @@ public boolean handle(int code) {
             // we've already logged this.
         }
 
-        if (loadSigar) {
-            // initialize sigar explicitly
-            try {
-                Sigar.load();
-                logger.trace("sigar libraries loaded successfully");
-            } catch (Throwable t) {
-                logger.trace("failed to load sigar libraries", t);
-            }
-        } else {
-            logger.trace("sigar not loaded, disabled via settings");
+        // Add hook to allow plugins to execute code before setting up security manager?
+        if (bootstrapPlugins) {
+            logger.trace("load plugins native stuff");
         }
```

Comment:
I think it might be the best tradeoff. In general the downside is it adds risk to load plugin classes before we start securitymanager. Today, no plugin code runs without security protection. We can continue to ban plugins from using features like JNI (in which case users will need to disable security alltogether), or we can try to support this stuff in a secure way. I like this idea because its simple, versus more complicated stuff like trying to give them special permissions.

I think its ok if we add something to the individual plugin configuration file (so lets move this to `setupPlugins()`), that indicates it needs this. Note that this would mean in the future such a plugin could never be reloaded. I think we should define a simple interface that passes in minimal parameters, so that such plugins can do this in a secure way (versus loading all their classes before security is enabled). So this means they specify a classname for it as the configuration value or whatever.

@s1monw opinions?